src/System.Globalization/tests/ISOWeekTests.cs

```diff
@@ -0,0 +1,215 @@
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Globalization
+{
+    public static class ISOWeekTests
+    {
+        // The following 71 years in a 400-year cycle have 53 weeks (371 days).
+        // Years not listed have 52 weeks (364 days). Add 2000 for current years.
+        private static readonly HashSet<int> s_YearsWith53Weeks = new HashSet<int>
+        {
+            004, 009, 015, 020, 026,
+            032, 037, 043, 048, 054,
+            060, 065, 071, 076, 082,
+            088, 093, 099, 105, 111, 116, 122,
+            128, 133, 139, 144, 150,
+            156, 161, 167, 172, 178,
+            184, 189, 195, 201, 207, 212, 218,
+            224, 229, 235, 240, 246,
+            252, 257, 263, 268, 274,
+            280, 285, 291, 296, 303, 308, 314,
+            320, 325, 331, 336, 342,
+            348, 353, 359, 364, 370,
+            376, 381, 387, 392, 398
+        };
+
+        // From https://en.wikipedia.org/wiki/ISO_week_date#Relation_with_the_Gregorian_calendar.
+        private static readonly DateData[] s_DateData =
+        {
+            new DateData("0001-01-01", "0001-W01-1"),
+            new DateData("2005-01-01", "2004-W53-6"),
+            new DateData("2005-01-02", "2004-W53-7"),
+            new DateData("2005-12-31", "2005-W52-6"),
+            new DateData("2006-01-01", "2005-W52-7"),
+            new DateData("2006-01-02", "2006-W01-1"),
+            new DateData("2006-12-31", "2006-W52-7"),
+            new DateData("2007-01-01", "2007-W01-1"),
+            new DateData("2007-12-30", "2007-W52-7"),
+            new DateData("2007-12-31", "2008-W01-1"),
+            new DateData("2008-01-01", "2008-W01-2"),
+            new DateData("2008-12-28", "2008-W52-7"),
+            new DateData("2008-12-29", "2009-W01-1"),
+            new DateData("2008-12-30", "2009-W01-2"),
+            new DateData("2008-12-31", "2009-W01-3"),
+            new DateData("2009-01-01", "2009-W01-4"),
+            new DateData("2009-12-31", "2009-W53-4"),
+            new DateData("2010-01-01", "2009-W53-5"),
+            new DateData("2010-01-02", "2009-W53-6"),
+            new DateData("2010-01-03", "2009-W53-7"),
+            new DateData("9999-12-31", "9999-W52-5"),
+        };
+
+        private static readonly YearData[] s_YearData =
+        {
+            new YearData(2005, "2005-01-03", "2006-01-01"),
+            new YearData(2006, "2006-01-02", "2006-12-31"),
+            new YearData(2007, "2007-01-01", "2007-12-30"),
+            new YearData(2008, "2007-12-31", "2008-12-28"),
+            new YearData(2009, "2008-12-29", "2010-01-03"),
+        };
+
+        public static IEnumerable<object[]> WeeksInYear_TestData()
+        {
+            for (int year = 0; year <= 400; year++)
+            {
+                bool isLong = s_YearsWith53Weeks.Contains(year);
+                yield return new object[] { 2000 + year, isLong };
+            }
```

Comment:
This one operation adds 400 unit tests that each show up individually, and it looks like the other theory inputs are similarly each adding a ton of inputs.  In the past we've had issues with too many tests bogging down the system, though that was when we were dealing with hundreds of thousands.  @Chrisboh, is that still a problem, such that we should consolidate all 400 of these inputs into one, or is it fine to have this?

Reply:
The method using this data can be just converted to a Fact and internally can loop on the data.

---
In reply to: [195476775](https://github.com/dotnet/corefx/pull/30386#discussion_r195476775) [](ancestors = 195476775)